Rearange documentation prioritizing more useful stuff higher (#50)
Lonami/pyndustric: README.md

A compiler from Python to Mindustry's assembly (logic programming language).

> The language **is** Python, so all simple programs you can imagine are possible. Check the [supported features][features]. There also might be some missing, so consider [contributing].
> 
> Beyond that, the
> supported "system calls" are documented in the [`pyndustri.pyi`] file.

Reviewer: ```suggestion
The language **is** Python, so all simple programs you can imagine are possible.
Check the [supported features][features] to learn what is currently possible.
If you find something is missing, you're welcome to [contribute]!

Beyond that, the supported "system calls" are documented in the [`pyndustri.pyi`] file.
```

To learn about the possible compiler errors, refer to the `ERR_` and `ERROR_DESCRIPTIONS`
constants in the [`constants.py`] file.

## Getting up and running

To install pyndustric as module run:

```sh
$ git clone https://github.com/Lonami/pyndustric.git

> $ pip install -e pyndustric

Reviewer: By the way I'm pretty sure it's possible to `pip install https://github.com/Lonami/pyndustric/archive/refs/heads/master.zip` directly, which does not even require `git` to be installed.

Also curious, you seem to use `install -e`, which is for "editable" installs.

```
  -e, --editable <path/url>   Install a project in editable mode (i.e. setuptools "develop mode") from a local project
                              path or a VCS url.
```

I don't think users who don't want to develop would want to do this, but maybe you had a reason for it.

Author: Reason I did it was because I just copied what I did not thinking much about it. Force of habit
I changed docs now

```

> Alternativly, you can run it as module without installation by placing your program file at the root of the project directory.

Reviewer: ```suggestion
Alternatively, you can run it as module without installation by placing your program file at the root of the project directory.
```

missed, implement support for more Python features, or write a peephole optimizer for the
compiler's output!

More info on contributing can be found in [`CONTRIBUTING.md`]

To run the tests, simply use the following command:

```sh
$ pip install -r dev-requirements.txt
$ pytest
```

[f-string]: https://docs.python.org/3/reference/lexical_analysis.html#f-strings
[ip-not-reset]: https://github.com/Anuken/Mindustry/issues/4189
[limit-k]: https://github.com/Anuken/Mindustry/blob/ab19e6ffbd7a64117cd70d3e3b88806c13822c94/core/src/mindustry/logic/LExecutor.java#L28
[`pyndustri.pyi`]: pyndustri.pyi
[`constants.py`]: pyndustric/constants.py
[`pyndustric`]: pyndustric
[features]: #Supported-features

> [contributing]: #Contributing

Reviewer: ```suggestion
[contribute]: #Contributing
```
